server/service/src/invoice/inbound_shipment/update/generate.rs

```diff
@@ -76,14 +84,30 @@ pub(crate) fn generate(
         None
     };
 
+    let tax_updated = tax_updated(&update_invoice.tax);
+    let update_tax_for_lines = if tax_updated {
+        Some(generate_tax_update_for_lines(
+            connection,
+            &update_invoice.id,
+            update_invoice.tax,
+        )?)
+    } else {
+        None
+    };
+
     Ok(GenerateResult {
         batches_to_update,
         empty_lines_to_trim: empty_lines_to_trim(connection, &existing_invoice, &patch.status)?,
         update_invoice,
         location_movements,
+        update_tax_for_lines,
     })
 }
 
+fn tax_updated(tax: &Option<f64>) -> bool {
+    tax.is_some()
```

Comment:
I would just inline it `tax.is_some()` is quite clear to...

Reply:
[cb099e7](https://github.com/openmsupply/open-msupply/pull/919/commits/cb099e728e50ec5073250d4cbd8bf1c82cefdb91)